node/src/service.rs

```diff
@@ -225,10 +225,6 @@ where
         bool,
     ) -> Result<Box<dyn ParachainConsensus<Block>>, Error>,
 {
-    if matches!(parachain_config.role, Role::Light) {
-        return Err("Light client not supported!".into());
```

Comment:
did you try running a light client?

Reply:
No I didn't. This is not us supporting a light client, but rather us removing the already unsupported `--light` CLI option

I believe the official light client is in this repo and the README explains how to try to set up a light client https://github.com/paritytech/smoldot. No idea if it would work

Reply:
I've been looking if it's still possible to set `Role::Light` to the config somehow, because if it is we can't remove this safeguard.

The Configuration type still has that enum field: https://github.com/Manta-Network/Manta/blob/b0095a5dc58b235987c31f32df71df7771be4515/node/src/service.rs#L355 

Substrate client can still set Role::Light https://github.com/paritytech/substrate/blob/1cca061ede75a4c693a3b71bd1b61d7275fe29e4/client/cli/src/commands/run_cmd.rs#L370

And we just delegate to substrate if i see this correctly:
https://github.com/Manta-Network/Manta/blob/b0095a5dc58b235987c31f32df71df7771be4515/node/src/command.rs#L636

Reply:
@ferrell-code looks like we need to keep this code around, otherwise the node will start with `--light` instead of exiting on the param

Reply:
I think you are looking at a different commit than polkadot-v0.9.28 (could be mistaken)

https://github.com/paritytech/substrate/blob/polkadot-v0.9.28/client/cli/src/commands/run_cmd.rs

no light client support in that file. They remove Light option in https://github.com/paritytech/substrate/pull/11865

Reply:
Oh ok, saw it. I checked out polkadot 9.28 but not substrate 9.28 :duck: